frontend/src/components/MapView/Layers/AnticipatoryActionFloodLayer/index.tsx

```diff
@@ -0,0 +1,204 @@
+    default:
+      return AAFloodColors.borderColors.notExceeded;
+  }
+};
+
+const CIRCLE_SIZE = 16;
+
+function AnticipatoryActionFloodLayer({
+  layer,
+}: AnticipatoryActionFloodLayerProps) {
+  // Load the layer default date if no date is selected
+  useDefaultDate(layer.id);
+  const { AAData } = useAnticipatoryAction(AnticipatoryAction.flood);
+  const { stations, stationSummaryData } = AAData;
+  const { startDate } = useSelector(dateRangeSelector);
+  const dispatch = useDispatch();
+
+  const handleFloodStationEvent = (stationName?: string) => {
+    dispatch(hidePopup()); // Hide any existing popups
+    if (stationName) {
+      dispatch(setAAFloodSelectedStation(stationName));
+    }
+  };
+
+  const selectedDateKey = startDate
+    ? new Date(startDate).toISOString().split('T')[0]
+    : null;
+
+  const floodStationsGeoJSON = useMemo(() => {
+    if (!stations.length) {
+      return null;
+    }
+
+    // Filter stations by selected date if available
+    const filteredStations = stations.filter((station: any) => {
+      if (!selectedDateKey) {
+        return true;
+      }
+      const avg = stationSummaryData?.[station.station_name];
+      const issueDate = avg?.forecast_issue_date
+        ? new Date(avg.forecast_issue_date).toISOString().split('T')[0]
+        : null;
+      return issueDate === selectedDateKey;
+    });
+
+    return {
+      type: 'FeatureCollection' as const,
+      features: filteredStations
+        .filter(
+          (station: any) =>
+            typeof station.longitude === 'number' &&
+            typeof station.latitude === 'number',
+        )
+        .map((station: any) => {
+          const avg = stationSummaryData?.[station.station_name];
+          if (!avg) {
+            return null;
+          }
+          return {
+            type: 'Feature' as const,
+            geometry: {
+              type: 'Point' as const,
+              coordinates: [station.longitude, station.latitude],
+            },
+            properties: {
+              station_name: station.station_name,
+              river_name: station.river_name,
+              station_id: station.station_id,
+              risk_level: avg.trigger_status || 'Not exceeded',
+              avg_discharge: 0,
+              max_discharge: 0,
+            },
+          };
+        }),
+    };
+  }, [stations, selectedDateKey, stationSummaryData]);
+
+  if (!floodStationsGeoJSON) {
+    return null;
+  }
+
+  const filteredStations = stations.filter((station: any) => {
+    if (!selectedDateKey) {
+      return true;
+    }
+    const avg = stationSummaryData?.[station.station_name];
+    const issueDate = avg?.forecast_issue_date
+      ? new Date(avg.forecast_issue_date).toISOString().split('T')[0]
+      : null;
+    return issueDate === selectedDateKey;
+  });
+
+  return (
+    <>
+      {filteredStations.map(station => {
+        if (
+          typeof station.longitude !== 'number' ||
```

Comment:
Better typing, if a station is listed here it should definitely have longitude and latitude